src/Processors/Formats/Impl/FreeformRowInputFormat.cpp

```diff
@@ -0,0 +1,473 @@
+#include "Processors/Formats/Impl/FreeformRowInputFormat.h"
+#include <DataTypes/DataTypeString.h>
+#include <IO/ReadBufferFromString.h>
+#include "Common/Exception.h"
+#include "Common/StringUtils/StringUtils.h"
+#include <Common/logger_useful.h>
+#include "Core/NamesAndTypes.h"
+#include "DataTypes/DataTypeNullable.h"
+#include "DataTypes/DataTypesNumber.h"
+#include "DataTypes/IDataType.h"
+#include "DataTypes/Serializations/ISerialization.h"
+#include "Formats/EscapingRuleUtils.h"
+#include "Formats/FormatFactory.h"
+#include "Formats/JSONUtils.h"
+#include "IO/ReadHelpers.h"
+#include "Processors/Formats/IRowInputFormat.h"
+#include "Processors/Formats/ISchemaReader.h"
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+    extern const int CANNOT_READ_ALL_DATA;
+    extern const int UNSUPPORTED_METHOD;
+}
+
+static inline void skipWhitespacesAndDelimiters(ReadBuffer & in)
+{
+    while (!in.eof() && (isWhitespaceASCII(*in.position()) || *in.position() == ',' || *in.position() == ':'))
+        ++in.position();
```

Comment:
I am not sure that we want to skip multiple `','` and `:` with possible spaces between them as one delimiter. For example in CSV format we can have empty field and `'abc,,def'`will represent 3 columns (maybe we can support it in Freedom format somehow?).
Also we can add more possible delimiters like `'|'`, `';'`, maybe some other punctuation symbols or their combinations (not sure 
Maybe do smth like this:
```
skipWhitespaceIfAny(in);
/// Skip possible delimiters like ',', '|', ':', etc or their combinations.
skipWhitespaceIfAny(in);
```

Reply:
Good point, I will try to experiment if we could really support empty fields, as the type we inferred from them will be null I assumed?